src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java

```diff
@@ -38,47 +46,101 @@
+
+            runProcess(new ProcessController(), gatk3Command.toArray(new String[gatk3Command.size()]));
+        } else {
+            System.out.println("Found precomputed gatk3Result");
+        }
+        final Path expectedResultsDir = Paths.get("expectedResults");
+        if ( !Files.exists(expectedResultsDir)) {
+            Files.createDirectory(expectedResultsDir);
+        }
+        Files.copy(gatk3Result.toPath(), expectedResultsDir.resolve(outputFile.getName()), StandardCopyOption.REPLACE_EXISTING);
+
+        assertGenotypesMatch(input, gatk3Result, extraArgs, reference);
+        assertVariantContextsMatch(input, gatk3Result, extraArgs, reference);
+    }
+
+
+    private static void runProcess(ProcessController processController, String[] command) {
+            final ProcessSettings prs = new ProcessSettings(command);
+            prs.getStderrSettings().printStandard(true);
+            prs.getStdoutSettings().printStandard(true);
+            final ProcessOutput output = processController.exec(prs);
+            Assert.assertEquals(output.getExitValue(), 0, "Process exited with non-zero value. Command: "+ Arrays.toString(command) + "\n");
+    }
+
     @Test(dataProvider = "gvcfsToGenotype")
-    public void testGenotypesOnly(String input, String expected, List<String> extraArgs) throws IOException {
-        assertGenotypesMatch(getTestFile(input), getTestFile(expected), extraArgs);
+    public void testGenotypesOnly(File input, File expected, List<String> extraArgs, String reference) throws IOException {
+        assertGenotypesMatch(input, expected, extraArgs, reference);
     }
 
     @Test(dataProvider = "gvcfsToGenotype")
-    public void testEntireVariantContext(String input, String expected, List<String> extraArgs) throws IOException {
-        assertVariantContextsMatch(getTestFile(input), getTestFile(expected), extraArgs);
+    public void testEntireVariantContext(File input, File expected, List<String> extraArgs, String reference) throws IOException {
+        assertVariantContextsMatch(input, expected, extraArgs, reference);
     }
 
-    private void assertVariantContextsMatch(File input, File expected, List<String> extraArgs) throws IOException {
+    private void assertVariantContextsMatch(File input, File expected, List<String> extraArgs, String reference) throws IOException {
         runGenotypeGVCFSAndAssertSomething(input, expected, extraArgs, (a, e) -> VariantContextTestUtils.assertVariantContextsAreEqual(a, e,
-                Arrays.asList("FS", //TODO There's a bug in GATK 3 computing FS
-                        "QD", //TODO QD has a cap value and anything that reaches that is randomized.  It's difficult to reproduce the same random numbers accross gatk3 -> 4
-                        "InbreedingCoeff")));
+                Arrays.asList("QD",//TODO QD has a cap value and anything that reaches that is randomized.  It's difficult to reproduce the same random numbers accross gatk3 -> 4
```

Comment:
I've extracted these to a named constant at the top to make it clear what these are.  They're excluded from comparison.